src/main/ApplicationImpl.cpp

```diff
@@ -661,4 +657,26 @@ ApplicationImpl::enabledInvariants() const
     }
     return result;
 }
+
+void
+ApplicationImpl::validateNetworkPassphrase()
+{
+    if (mConfig.NETWORK_PASSPHRASE.empty())
+    {
+        throw std::invalid_argument("NETWORK_PASSPHRASE not configured");
+    }
+
+    std::string prevNetworkPassphrase =
+        mPersistentState->getState(PersistentState::kNetworkPassphrase);
+    if (prevNetworkPassphrase.empty())
+    {
+        mPersistentState->setState(PersistentState::kNetworkPassphrase,
+                                   mConfig.NETWORK_PASSPHRASE);
+    }
+    else if (mConfig.NETWORK_PASSPHRASE != prevNetworkPassphrase)
+    {
+        throw std::invalid_argument("NETWORK_PASSPHRASE does not match previous"
```

Comment:
Including the values would be useful, so that the operator knows right away what kind of mistake was made